docs/contracts.rst

```diff
@@ -953,6 +953,29 @@ not known in the context of the class where it is used,
 although its type is known. This is similar for ordinary
 virtual method lookup.
 
+.. index:: ! constructor
+
+Constructors
+============
+Constructor is an optional function with the same name as the contract which is executed upon contract creation. 
+Constructor functions can be either ``public`` or ``internal``.
+
+::
+
```

Comment:
verison pragma is missing here

Reply:
@axic for the version pragma, do we specify the minimum version required to run the sample or simply the current version, i.e. `0.4.19`?

Reply:
So far the rule has been the minimum required by the example.

Reply:
The release notes aren't very clear on when this was introduced but Remix thinks this breaks from 0.4.10 downwards so I'll use pragma min 0.4.11 in the example. 
From release notes on 0.4.11:
>Type system: Contract inheriting from base with unimplemented constructor should be abstract.